Storage/ThirdParty/SwiftData.swift

```diff
@@ -104,7 +104,7 @@ public class SwiftData {
      * close a database connection and run a block of code inside it.
      */
     func withConnection(flags: SwiftData.Flags, synchronous: Bool=true, cb: (db: SQLiteDBConnection) -> NSError?) -> NSError? {
-        let conn: ConcreteSQLiteDBConnection?
+        weak var conn: ConcreteSQLiteDBConnection?
```

Comment:
This isn't sufficient. Almost immediately after we call `getSharedConnection` and we assign it here, we hit a `guard let` block that takes a strong reference. This weak hold only allows us to close the DB between `getSharedConnection` and using it.

Down on 140 and 146 we're passing in the strong reference, captured in a closure, to `dispatch_async`.

In order to do what you're trying to do here, you need those dispatch blocks to have a `weak` capture list for `connection`.

You also need to invert the rest of this method: the `guard let connection = conn…` stuff needs to happen _inside_ the dispatch block, because that's the point at which we might find the connection to be gone.